Cython/Compiler/PyrexTypes.py

```diff
@@ -4663,7 +4667,7 @@ def best_match(arg_types, functions, pos=None, env=None, args=None):
             candidates.append((func, func_type))
 
     # Optimize the most common case of no overloading...
-    if len(candidates) == 1:
+    if len(candidates) == 1 and not validate_types_fully:
```

Comment:
This looks funny and seems worth explaining – why continue if we only have a single candidate anyway?

Reply:
That's gone now, but I think the answer is "because we might actually have no candidates, but just not realize it"